### PR TITLE
Color generator

### PR DESCRIPTION
**Pull request by luca992**
add the color generator in the original project back in ported to kotlin

also I updated the android sdk, kotlin, and gradle

### CODE REVIEW

**Review by Kennyc1012 — CHANGES_REQUESTED** [review]
Rebase your change onto master which removes the need for the sdk update commit

File: library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package com.kennyc.textdrawable
+
+import kotlin.random.Random
+
+class ColorGenerator private constructor(private val mColors: List<Int>) {
+    private val mRandom: Random = Random(System.currentTimeMillis())
```
Comment (Kennyc1012) [inline library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt]: Do not use Hungarian notation, should  just be `random: Random`

File: library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    val randomColor: Int
+        get() = mColors[mRandom.nextInt(mColors.size)]
```
Comment (Kennyc1012) [inline library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt]: Same here

File: library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    fun getColor(key: Any): Int {
+        return mColors[Math.abs(key.hashCode()) % mColors.size]
```
Comment (Kennyc1012) [inline library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt]: Same here

File: library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    companion object {
+
+        var DEFAULT: ColorGenerator = create(listOf(
```
Comment (Kennyc1012) [inline library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt]: Should be a val, not var

File: library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                -0x65bc2,
+                -0x1b39d2,
+                -0x98408c,
+                -0xa65d42,
+                -0xdf6c33,
+                -0x529d59,
+                -0x7fa87f
+        ))
+
+        var MATERIAL: ColorGenerator = create(listOf(
```
Comment (Kennyc1012) [inline library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt]: Same here

File: library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package com.kennyc.textdrawable
+
+import kotlin.random.Random
+
+class ColorGenerator private constructor(private val mColors: List<Int>) {
```
Comment (Kennyc1012) [inline library/src/main/java/com/kennyc/textdrawable/ColorGenerator.kt]: Do not use Hungarian notation, should be `colors: List<Int>`

**Review by Kennyc1012 — CHANGES_REQUESTED** [review]

**Comment by luca992** [comment]
Made the fixes and rebased it onto master

**Pull request merged**